pkg/apis/pipeline/v1beta1/pipeline_validation.go

```diff
@@ -546,3 +549,14 @@ func validateGraph(tasks []PipelineTask) *apis.FieldError {
 	}
 	return nil
 }
+
+func validateMatrix(ctx context.Context, tasks []PipelineTask) (errs *apis.FieldError) {
```

Comment:
can we instead implement this as a member function of `(pt PipelineTask)`?

Reply:
sure, refactored it :)